src/opnsense/mvc/app/models/OPNsense/Monit/Monit.php

```diff
@@ -77,26 +167,44 @@ public function performValidation($validateFullModel = false)
                         // test node validations
                         switch ($node->getInternalXMLTagName()) {
                             case 'type':
+                                // only 'Custom' is allowed if test is linked to a service
                                 $testUuid = $parentNode->getAttribute('uuid');
-                                if ($node->isFieldChanged() && $this->isTestServiceRelated($testUuid)) {
+                                if (strcmp((string)$node, 'Custom') != 0 &&
```

Comment:
@fbrendel I'm a bit struggling why this is checked, if it's custom I can move it to something else, when faulty I can't move back. We probably should keep it locked for all, since migration fixes the default types.

Reply:
The Custom type is meant as a fallback because I'm not sure if getTestType can successfully parse all (valid) variations of the condition string.
Maybe we should remove Custom and print an error message instead if getTestType fails?

Reply:
@fbrendel I've removed it :) if it looks safe enough, we could leave it like this and see what happens if anyone complains